src/EditorFeatures/CSharpTest/CodeActions/LambdaSimplifier/LambdaSimplifierTests.cs

```diff
@@ -557,6 +557,51 @@ static void Main()
 }",
 @"using System;
 
+class Program
+{
+    static void Main()
+    {
+        Action a = Console.WriteLine;
+    }
+}");
+        }
+
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsLambdaSimplifier)]
+        public async Task TestMissingCaretPositionInside()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class Program
+{
+    static void Main()
+    {
+        Action a = () => {
+            Console.[||]WriteLine();
```

Comment:
This is a breaking change, this one would pass previously. Even though I don't think it should.

Reply:
The issue doesn't exist for ChangeTypeToConcrete/Var because that one has one additional check for intersection with Type.

Reply:
cc: @CyrusNajmabadi

Reply:
There is even a VB test that tests for that: 
https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/EditorFeatures/VisualBasicTest/ConvertAutoPropertyToFullProperty/ConvertAutoPropertyToFullPropertyTests.vb#L121

Reply:
The issue (highlighted by the VB test) is actually slightly bigger. The refactoring registers for
`pu[||]blic int Asdf2 { get; set; }` but not `public i[||]nt Asdf2 { get; set; }` because `public` is Token with immediate parent of the Token Declaration but int is a PredefinedTypeNode…

How much does this bother us?

Reply:
Also there's already some precedence of enabling refactorings for caret in type e.g.: Change to concrete type (var -> type) and vice versa. (It explicitly makes sense there however)

Reply:
Please review test change, don't want this change to slip review. @CyrusNajmabadi  or @mavasani

Reply:
I believe the new test behavior is reasonable.

Reply:
Yes.  thie behavior is reasonable.  It's worth nothing:

1. this feature hasn't shipped. :)
2. this feature was written probably 5+ years ago, and didn't have the benefit of all our learnings about good refactoring selections.

So it's really reasonable to make changes here as long as they're intentional and seem reasonable.

Reply:
Ok.